In [10]:
import pandas as pd
import numpy as np
import os

DATA_DIR = "../data/raw/"

train_df = pd.read_csv(os.path.join(DATA_DIR, "train.csv"))
test_df  = pd.read_csv(os.path.join(DATA_DIR, "test.csv"))

print("Train shape:", train_df.shape)
print("Test shape:", test_df.shape)

train_df.head()

Train shape: (891, 12)
Test shape: (418, 11)


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [11]:

print("=== TRAIN INFO ===")
train_df.info()

print("\n=== TEST INFO ===")
test_df.info()

=== TRAIN INFO ===
<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB

=== TEST INFO ===
<class 'pandas.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 

In [12]:

print("=== TRAIN: Missing ===")
m_train = train_df.isnull().sum()
p_train = (m_train / len(train_df) * 100).round(2)
print(pd.DataFrame({"missing": m_train, "pct": p_train}))

print("\n=== TEST: Missing ===")
m_test = test_df.isnull().sum()
p_test = (m_test / len(test_df) * 100).round(2)
print(pd.DataFrame({"missing": m_test, "pct": p_test}))

=== TRAIN: Missing ===
             missing    pct
PassengerId        0   0.00
Survived           0   0.00
Pclass             0   0.00
Name               0   0.00
Sex                0   0.00
Age              177  19.87
SibSp              0   0.00
Parch              0   0.00
Ticket             0   0.00
Fare               0   0.00
Cabin            687  77.10
Embarked           2   0.22

=== TEST: Missing ===
             missing    pct
PassengerId        0   0.00
Pclass             0   0.00
Name               0   0.00
Sex                0   0.00
Age               86  20.57
SibSp              0   0.00
Parch              0   0.00
Ticket             0   0.00
Fare               1   0.24
Cabin            327  78.23
Embarked           0   0.00


In [13]:
print("=== TRAIN describe() ===")
print(train_df.describe())

print("\n=== TEST describe() ===")
print(test_df.describe())

=== TRAIN describe() ===
       PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200  

=== TEST describe() ===
       PassengerId      Pclass 

In [14]:
for col in ["Pclass","Sex","Embarked","Survived"]:
    print(f"\n=== {col} ===")
    print(train_df[col].value_counts(dropna=False))


=== Pclass ===
Pclass
3    491
1    216
2    184
Name: count, dtype: int64

=== Sex ===
Sex
male      577
female    314
Name: count, dtype: int64

=== Embarked ===
Embarked
S      644
C      168
Q       77
NaN      2
Name: count, dtype: int64

=== Survived ===
Survived
0    549
1    342
Name: count, dtype: int64


# Day 1 — Data Loading and Initial Analysis

## Goal
Perform a first-look audit of the Titanic dataset before any modeling.

## Data
- `train.csv` — 891 rows, 12 columns (includes target `Survived`)
- `test.csv` — 418 rows, 11 columns (no target)

## What We Checked
1. **Shape** — train (891, 12), test (418, 11)
2. **Dtypes & non-null counts** — `.info()`
3. **Missing values** — counts + percentages
4. **Numeric statistics** — `.describe()`
5. **Categorical counts** — `.value_counts(dropna=False)`

## Key Findings
- **Missing values (train):**
  - `Age` — ~20% missing
  - `Cabin` — ~77% missing
  - `Embarked` — ~0.22% missing (2 rows)
- **Target balance:** 61.6% died, 38.4% survived (imbalanced)
- **Key categories:**
  - `Sex` — 577 male, 314 female
  - `Pclass` — 491 in 3rd, 216 in 1st, 184 in 2nd
  - `Embarked` — 644 from S, 168 from C, 77 from Q

## Implications for Later Days
- `Age` needs imputation (Day 3).
- `Cabin` is too sparse → transform to binary "HasCabin" or drop (Day 3).
- Class imbalance → track precision/recall, not just accuracy (Day 5).
